### PR TITLE
Amazon Test using POM and FluentInterface

### PR DESCRIPTION
**Pull request by AndrewHat**
I did the Amazon Search Test to verify what was written with the top item found

### CODE REVIEW

File: src/main/java/HomePage.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    public HomePage writeSearchBar(String text) {
+        WebElement textSearchBarElement= driver.findElement(By.id("twotabsearchtextbox"));
+        textSearchBarElement.sendKeys(text);
+
+        return this;
+    }
+
+    public SearchResultPage enterSearchSubmitButton() {
```
Comment (barranquerox) [inline src/main/java/HomePage.java:19]: what does this mean? is this english??

File: src/main/java/HomePage.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+public class HomePage {
+    WebDriver driver;
+
+    public HomePage(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public HomePage writeSearchBar(String text) {
+        WebElement textSearchBarElement= driver.findElement(By.id("twotabsearchtextbox"));
```
Comment (barranquerox) [inline src/main/java/HomePage.java:13]: can you extract the selector in a class member?

File: src/main/java/HomePage.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        textSearchBarElement.sendKeys(text);
+
+        return this;
+    }
+
+    public SearchResultPage enterSearchSubmitButton() {
+        WebElement buttonElement = driver.findElement(By.id("nav-search-submit-button"));
```
Comment (barranquerox) [inline src/main/java/HomePage.java:20]: Selectors in a class member please

File: src/main/java/ItemPage.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+public class ItemPage {
+    WebDriver driver;
+
+    public ItemPage(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public Boolean containsProductTitleResult(String expectedText) {
+        WebElement titleElement = driver.findElement(By.id("productTitle"));
+        boolean containsTitle = titleElement.getText().contains(expectedText);
```
Comment (barranquerox) [inline src/main/java/ItemPage.java:14]: The contains goes in the test, not in the  page object.

File: src/main/java/SearchResultPage.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+public class SearchResultPage {
+    WebDriver driver;
+
+    public SearchResultPage(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public ItemPage openFirstResult() {
+        By firstResultSelector = By.cssSelector("[cel_widget_id=\"MAIN-SEARCH_RESULTS-3\"] .s-image");
```
Comment (barranquerox) [inline src/main/java/SearchResultPage.java:13]: Can you create a generic function of the type: `openResult(int index)` instead of `openFirstResult`

**Review by barranquerox — CHANGES_REQUESTED** [review]

File: src/main/java/HomePage.java (hunk 1 above)
Comment (AndrewHat) [reply src/main/java/HomePage.java:19]: I thought that the use of the button was a requirement

File: src/main/java/ItemPage.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    By titleSelector = By.id("productTitle");
+
+    public ItemPage(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public String getProductTitle() {
+        WebElement titleElement = driver.findElement(titleSelector);
+
+        return titleElement.getText();
```
Comment (barranquerox) [inline src/main/java/ItemPage.java:16]: why not just do in one line:
`return driver.findElement(titleSelector).getText();` ?

**Review by barranquerox — CHANGES_REQUESTED** [review]

**Review by barranquerox — APPROVED** [review]

**Pull request merged**